recipes/gw/meta.yaml

```diff
@@ -12,48 +12,71 @@ source:
     - fix-linker-issues.patch  # [linux]
 
 build:
-  number: 2
+  number: 3
   run_exports:
     - {{ pin_subpackage('gw', max_pin="x") }}
 
 requirements:
   build:
     - make
+    - {{ compiler('c') }}
     - {{ compiler('cxx') }}
-    # - sysroot_linux-64 >=2.17
-    - {{ cdt('mesa-libgl-devel') }}   # [linux]
-    - {{ cdt('mesa-libegl-devel') }}  # [linux]
-    - {{ cdt('mesa-dri-drivers') }}   # [linux]
-    - {{ cdt('libselinux') }}         # [linux]
-    - {{ cdt('libxdamage') }}         # [linux]
-    - {{ cdt('libxxf86vm') }}         # [linux]
-    - {{ cdt('libxext') }}            # [linux]
-
+    - {{ cdt('xorg-x11-proto-devel') }}  # [linux]
+    - {{ cdt('libx11-devel') }}          # [linux]
+    - {{ cdt('libxext-devel') }}         # [linux]
+    - {{ cdt('libxrender-devel') }}      # [linux]
+    - {{ cdt('mesa-libgl-devel') }}      # [linux]
+    - {{ cdt('mesa-libegl-devel') }}     # [linux]
+    - {{ cdt('mesa-dri-drivers') }}      # [linux]
+    - {{ cdt('mesa-libgbm') }}           # [linux]
+    - {{ cdt('libxcb') }}                # [linux]
+    - {{ cdt('libxau-devel') }}          # [linux]
+    - {{ cdt('libdrm-devel') }}          # [linux]
+    - {{ cdt('libxcomposite-devel') }}   # [linux]
+    - {{ cdt('libxcursor-devel') }}      # [linux]
+    - {{ cdt('libxi-devel') }}           # [linux]
+    - {{ cdt('libxrandr-devel') }}       # [linux]
+    - {{ cdt('libxtst-devel') }}         # [linux]
+    - {{ cdt('libselinux-devel') }}      # [linux]
+    - {{ cdt('libxdamage') }}            # [linux]
+    - {{ cdt('libxdamage-devel') }}      # [linux]
+    - {{ cdt('libxfixes') }}             # [linux]
+    - {{ cdt('libxfixes-devel') }}       # [linux]
+    - {{ cdt('libxxf86vm') }}            # [linux]
+    - {{ cdt('libxrandr') }}             # [linux]
+    - {{ cdt('xorg-x11-server-xvfb') }}  # [linux]
+    - {{ cdt('libxt-devel') }}           # [linux]
+    - {{ cdt('libsm-devel') }}           # [linux]
+    - {{ cdt('libice-devel') }}          # [linux]
+    - {{ cdt('libuuid-devel') }}         # [linux]
+    - {{ cdt('libglvnd-glx') }}          # [linux]
+    - {{ cdt('libglvnd-egl') }}          # [linux]
+    - {{ cdt('libglvnd-devel') }}        # [linux]
```

Comment:
I am unsure whether this is correct. As far as I know, conda-forge, unlike the default channel, nowadays offers gl and x11 packages. Hence, they should not be marked as cdt I think. @mbargull what do you think?

Reply:
It would be preferable not to use the CDT I think, but no idea how the non-cdt packages should be linked properly. When they are installed they doesn't end up in conda/lib

Reply:
Also, other conda forge packages such as Qt still use CDT https://github.com/conda-forge/qt-feedstock/blob/main/recipe/meta.yaml

Reply:
`libglvnd` has just been added to conda-forge in the last couple of days.
I'm not much up to date on everything, but I know discussions are happening at https://github.com/conda-forge/cdt-builds/issues/66#issuecomment-2284266504 , https://github.com/conda-forge/qt-main-feedstock/pull/284 .

Reply:
As for the xorg packages, you should be able to add most of them to `requirements/host` instead of the CDT ones in `requirements/build`.
Some of those do, but some don't have `run_exports`; for those that do have `run_exports` defined, you don't have to explicitly list them in `requirements/run` but for those that don't you should add them, of course.

Reply:
IMO, you should avoid linking to the CDT ones if you can use/are using the Conda packages as run time dependencies.

Reply:
> When they are installed they don't end up in conda/lib

What do you mean with "conda/lib"? And for which packages in particular?

Reply:
Thanks @mbargull, I wasn't aware of the libglvnd update, it looks like a openGL can be linked then - I will have a go a using that package. Previously I was installing the conda-forge Mesa packages, but I didn't understand how to link them as they were not in a visible place - perhaps pkg-config is suppose to be used or something?

Reply:
I suggest using libegl-devel, libgl-devel, libglx-devel... in the host dependencies. This will automatically add a libegl, libgl, libglx... run dependency without the need to specify them. This is not a CDT so no encapsulation into jinja.